### PR TITLE
feat(deref): Impl `Deref` for `FractionLike`

### PR DESCRIPTION
**Pull request by malik672**
fix issue #41 
fix issue #42

### CODE REVIEW

File: src/entities/fractions/fraction.rs [hunk 1]
```diff
@@ -20,7 +20,15 @@ impl<M: Default> Default for FractionLike<M> {
...
 }
 
-// Type alias for a simple Fraction without metadata
+impl Deref for FractionLike<Currency> {
+    type Target = Currency;
+
+    fn deref(&self) -> &Self::Target {
+        &self.meta
+    }
+}
```
Comment (shuhuiluo) [inline src/entities/fractions/fraction.rs]:
```suggestion
impl<M> Deref for FractionLike<M> {
    type Target = M;

    fn deref(&self) -> &Self::Target {
        &self.meta
    }
}
```
Do the same for `CurrencyLike<M>`. Refactor all occurrences of `.meta.`.

File: src/constants.rs [hunk 2]
```diff
@@ -1,4 +1,5 @@
 use crate::prelude::*;
+///```
```
Comment (shuhuiluo) [inline src/constants.rs]:
```suggestion
```

File: src/entities/fractions/currency_amount.rs [hunk 3]
```diff
@@ -1,19 +1,21 @@
...
-// Struct representing metadata about a currency
+/// Struct representing metadata about a currency
 #[derive(Clone, Debug, PartialEq)]
 pub struct CurrencyMeta<T: CurrencyTrait> {
     pub currency: T,
     pub decimal_scale: BigUint,
 }
 
-// Implementation of methods for CurrencyAmount
+/// Implementation of methods for CurrencyAmount
```
Comment (shuhuiluo) [inline src/entities/fractions/currency_amount.rs:16]:
```suggestion
```
Unnecessary.

File: src/entities/currency.rs [hunk 4]
```diff
@@ -24,14 +24,17 @@ pub trait CurrencyTrait: BaseCurrency {
     fn wrapped(&self) -> Token;
 }
 
+///implementation of CurrencyTrait for a currency
```
Comment (shuhuiluo) [inline src/entities/currency.rs]:
```suggestion
```
Unnecessary.

**Review by shuhuiluo — CHANGES_REQUESTED** [review]

**Review by shuhuiluo — CHANGES_REQUESTED** [review]
Impl `Deref` for `CurrencyLike`.

**Comment by malik672** [comment]
> Impl `Deref` for `CurrencyLike`.

will have to merge this first then pull into my pc

**Review by shuhuiluo — APPROVED** [review]

**Pull request merged**